src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1518,6 +1518,53 @@ private static bool CanDoGraphAutoLayout(object o)
             return true;
         }
 
+        private void ShowAllWires(object o)
+        {
+            var nodeModels = DynamoSelection.Instance.Selection.OfType<NodeModel>().Where(n => n.AllConnectors.Any(x => x.IsHidden)).ToList();
+            ShowHideAllWires(nodeModels, false);
+        }
+
+        private bool CanShowAllWires(object o)
+        {
+            return DynamoSelection.Instance.Selection.OfType<NodeModel>()
+                .Any(n => n.AllConnectors.Any(x => x.IsHidden));
+        }
+
+        private void HideAllWires(object o)
+        {
+            var nodeModels = DynamoSelection.Instance.Selection.OfType<NodeModel>().Where(n => n.AllConnectors.Any(x => !x.IsHidden)).ToList();
+            ShowHideAllWires(nodeModels, true);
+
+        }
+
+        private bool CanHideAllWires(object o)
+        {
+            return DynamoSelection.Instance.Selection.OfType<NodeModel>()
+                .Any(n => n.AllConnectors.Any(x => !x.IsHidden));
+        }
+
+        /// <summary>
+        /// Shows or Hides all wires of a list of nodeModels
+        /// </summary>
+        /// <param name="nodeModels"></param>
+        /// <param name="isHidden"></param>
+        private void ShowHideAllWires(List<NodeModel> nodeModels, bool isHidden)
+        {
+            if (!nodeModels.Any()) return;
+
+            foreach (var nodeModel in nodeModels)
+            {
+                var connectors = nodeModel.AllConnectors;
+                var enumerator = connectors.GetEnumerator();
+                while (enumerator.MoveNext())
```

Comment:
I was thinking may be using an enumerator for connectors list could be slow and could affect graphs with lot of connectors. Can we just iterate the list normally?